alf/agents/a2c/a2c_agent_test.py

```diff
@@ -0,0 +1,175 @@
+            state_spec=state_spec,
+            name='DummyActorNet')
+        single_action_spec = tf.nest.flatten(output_tensor_spec)[0]
+        activation_fn = None if unbounded_actions else tf.nn.tanh
+        self._output_tensor_spec = output_tensor_spec
+        self._layers = [
+            tf.keras.layers.Dense(
+                single_action_spec.shape.num_elements() * 2,
+                activation=activation_fn,
+                kernel_initializer=tf.compat.v1.initializers.constant(
+                    [[2, 1], [1, 1]]),
+                bias_initializer=tf.compat.v1.initializers.constant(5),
+            ), ]
+
+    def call(self, observations, step_type=None, network_state=()):
+        states = tf.cast(tf.nest.flatten(observations)[0], tf.float32)
+        for layer in self.layers:
+            states = layer(states)
+
+        single_action_spec = tf.nest.flatten(self._output_tensor_spec)[0]
+        actions, stdevs = tf.split(states, 2, axis=1)
+        actions = tf.reshape(actions, [-1] + single_action_spec.shape.as_list())
+        stdevs = tf.reshape(stdevs, [-1] + single_action_spec.shape.as_list())
+        actions = tf.nest.pack_sequence_as(self._output_tensor_spec, [actions])
+        stdevs = tf.nest.pack_sequence_as(self._output_tensor_spec, [stdevs])
+
+        distribution = nest.map_structure_up_to(
+            self._output_tensor_spec, tfp.distributions.Normal, actions, stdevs)
+        return distribution, network_state
+
+
+class DummyValueNet(network.Network):
+
+    def __init__(self, observation_spec, name=None, outer_rank=1):
+        super(DummyValueNet, self).__init__(observation_spec, (), 'DummyValueNet')
+        self._outer_rank = outer_rank
+        self._layers.append(
+            tf.keras.layers.Dense(
+                1,
+                kernel_initializer=tf.compat.v1.initializers.constant([2, 1]),
+                bias_initializer=tf.compat.v1.initializers.constant([5])))
+
+    def call(self, inputs, unused_step_type=None, network_state=()):
+        hidden_state = tf.cast(tf.nest.flatten(inputs), tf.float32)[0]
+        batch_squash = network_utils.BatchSquash(self._outer_rank)
+        hidden_state = batch_squash.flatten(hidden_state)
+        for layer in self.layers:
+            hidden_state = layer(hidden_state)
+        value_pred = tf.squeeze(batch_squash.unflatten(hidden_state), axis=-1)
+        return value_pred, network_state
+
+
+class A2cAgentTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        super(A2cAgentTest, self).setUp()
+        tf.compat.v1.enable_resource_variables()
+        self._obs_spec = tensor_spec.TensorSpec([2], tf.float32)
+        self._time_step_spec = ts.time_step_spec(self._obs_spec)
+        self._action_spec = tensor_spec.BoundedTensorSpec([1], tf.float32, -1, 1)
+
+    def test_create_agent(self):
+        a2c_agent.A2CAgent(
+            self._time_step_spec,
+            self._action_spec,
+            actor_net=DummyActorNet(
```

Comment:
Why do you need to have DummyActorNet and DummyValueNet instead of using ActorDistributionNetwork and ValueNetwork for this test?

Reply:
agent tests in TFAgents using DummyNetworks . i just follow it , i do not know why too .